### PR TITLE
[BC] New storage service architecture & flysystem integration

### PR DESCRIPTION
**Pull request by adriangornypl**

### CODE REVIEW

**Review by naros1988 — APPROVED** [review]

File: Dto/UploadedFileDto.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace AppVerk\GoogleCloudStorageMediaBundle\Dto;
```
Comment (michaljedraszczyk) [inline Dto/UploadedFileDto.php]: Please change workspace from Dto to Model

File: Dto/UploadedFileDto.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+class UploadedFileDto {
```
Comment (michaljedraszczyk) [inline Dto/UploadedFileDto.php]: Simple UploadFile as name

File: Flysystem/Retriever/LocalObjectUrlRetriever.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+use Google\Cloud\Storage\StorageClient;
+use Google\Cloud\Storage\StorageObject;
+
+class LocalObjectUrlRetriever implements UrlRetrieverInterface
+{
+    public function getUrl(string $filename): string
+    {
+        return $filename;
+    }
+}
```
Comment (michaljedraszczyk) [inline Flysystem/Retriever/LocalObjectUrlRetriever.php]: make sure all files have empty line on the end: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

File: Service/MediaUploader.php [hunk 4]
```diff
@@ -113,7 +114,7 @@ private function validate(UploadedFile $file, ?string $groupName = null): void
         $maxSize = $this->mediaValidation->getMaxSize($groupName);
         if ($maxSize) {
             if (!($fileSize = $file->getClientSize())) {
-                throw new NotFoundHttpException();
+                throw new InvalidSizeException('', null, 404);
```
Comment (michaljedraszczyk) [inline Service/MediaUploader.php]: Add custom message what goes wrong.

File: Service/MediaValidation.php [hunk 5]
```diff
@@ -80,10 +80,16 @@ public function getMaxSize(?string $groupName = null): ?int
 
     public function getGroupSizes(?string $groupName = null): array
     {
+        if (!$groupName) {
+            return [];
+        }
+
         $group = $this->getGroup($groupName);
+
         if (!$group) {
```
Comment (michaljedraszczyk) [inline Service/MediaValidation.php]: merge that to one statement.

File: Service/v2/StorageService.php [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    public function sanitize(string $filename): string
+    {
+        return preg_replace(['/\s/', '/\.[\.]+/', '/[^\w_\.\-]/'], ['_', '.', ''], $filename);
+    }
+
+    public function remove(string $filename): bool
+    {
+        try {
+            if (true === $this->filesystem->fileExists($filename)) {
+                $this->filesystem->delete($filename);
```
Comment (michaljedraszczyk) [inline Service/v2/StorageService.php:94]: Lets add new events for add, remove. Will be needed for example for external cropping/resizing.

**Review by michaljedraszczyk — CHANGES_REQUESTED** [review]

File: Dto/UploadedFileDto.php (hunk 1 above)
Comment (adriangornypl) [reply Dto/UploadedFileDto.php]: Done

File: Dto/UploadedFileDto.php (hunk 2 above)
Comment (adriangornypl) [reply Dto/UploadedFileDto.php]: Done, refactored ?string to string in most cases

File: Service/MediaUploader.php (hunk 4 above)
Comment (adriangornypl) [reply Service/MediaUploader.php]: Done

File: Service/MediaValidation.php (hunk 5 above)
Comment (adriangornypl) [reply Service/MediaValidation.php]: Done

File: Service/v2/StorageService.php (hunk 6 above)
Comment (adriangornypl) [reply Service/v2/StorageService.php:94]: Added

**Pull request merged**